common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -355,6 +373,30 @@ int addHoverEvent(HoverEvent hoverEvent) {
         return hoverEvents.size();
     }
 
+    public int getWidth() {
+        return McUtils.mc().font.width(getString(PartStyle.StyleType.NONE));
+    }
+
+    public StyledText[] split(String s) {
+        return Arrays.stream(getString().split(s)).map(StyledText::fromString).toArray(StyledText[]::new);
+    }
+
+    public StyledText subtext(int start, int end) {
+        return StyledText.fromString(getString().substring(start, end));
+    }
+
+    public StyledText subtext(int start) {
+        return StyledText.fromString(getString().substring(start));
+    }
```

Comment:
I don't like these kinds of operations in StyledText, just like `split`, this is not the way to do things. Let's remove this and find an alternative way for their usages.

Reply:
what do we do here then:
```java
private StyledText getBaseName(StyledText name) {
        int bracketIndex = name.getString().lastIndexOf('[');
        return bracketIndex == -1 ? name : name.subtext(0, bracketIndex);
    }
```

Reply:
This can easily implemented with StyledText. Most likely the `Item Name [97%]` will be 2 parts: `Item Name` and `[97%]`. I just have to think about how to not make it look weird (we would have to use iterateBackwards as of right now).

Again, use getString and I will worry about it later.